src/data_type/mod.rs

```diff
@@ -2296,7 +2312,7 @@ impl Variant for DataType {
                 match (self, other) {
                     // If self and other are from different variants
                     (DataType::Null, _) => true, // Any element of self is also in other
-                    (DataType::Unit(_), DataType::Unit(_)) => true,
+                    (DataType::Unit(_), DataType::Unit(_)) | (DataType::Unit(_), DataType::Optional(_))=> true,
```

Comment:
Can you make sure the formatting is correct (space before =>)
Run `cargo fmt` before you commit.